lib/assert.js

```diff
@@ -438,6 +433,9 @@ function expectedException(actual, expected, msg) {
     // as well.
     if (expected instanceof Error) {
       keys.push('name', 'message');
+    } else if (keys.length === 0) {
+      throw new ERR_INVALID_ARG_VALUE('error',
+                                      expected, 'may not be an empty object');
```

Comment:
is there any background on why this constraint was added?

It seems perfectly valid to me to do `var sentinel = {}; assert.throws(() => { throw sentinel; }, sentinel, 'throws the sentinel');`, for example.

Reply:
Before this change, nothing would have been validated as the signature was about validating the properties of the object and there are no properties to validate.

Reply:
ahh ok, so there's no way to validate that a specific thing with identity is thrown?

Reply:
It's possible with the function validation: `var sentinel = {}; assert.throws(() => { throw sentinel; }, (error) => assert(error === sentinel, 'Sentinel should be thrown'));`